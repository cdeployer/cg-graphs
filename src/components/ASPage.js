import React, { Component } from 'react';
import { connect } from 'react-redux';
import ASPerformance from './ASPerformance';


export class ASPage extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        {Object.keys(this.props.views).map(key => {
          return (
            <div key={key} className="row">
              <div className="col-md-6 panel panel-default">
                <ASPerformance view={this.props.views[key]} viewId={key}/>
              </div>
            </div>
          )
        })}
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    views: state.ASPerformanceViews
  }
}


export default connect(mapStateToProps)(ASPage)
