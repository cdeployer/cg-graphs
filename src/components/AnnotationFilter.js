// import React, {
//   Component
// } from 'react';
// import {
//   connect
// } from 'react-redux'
// import PlotlyGraph from './Plot.js';
// import Select from 'react-select';
// import update from 'react/lib/update'
// import Highlighter from 'react-highlight-words'
// import 'react-select/dist/react-select.css';
// //need to move file to this location
// //import '../api/publicAnnotation.json';
// import {
//   countryIsSelected,
//   fetchDataIfNeeded,
//   getCountryRanking,
//   changeMeasure
// } from '../actions/cubeActions';
//
// // Filter by country and risk
// let annotation_dates = [];
// let annotation_notes = [];
// let annotations = [];
// fetch(`/static/scripts/publicAnnotation.json`)
//   .then((response) => {
//     return response.json()
//   })
//   .then((annFilterByCountry) => {
//       for (var ann_num in annFilterByCountry.notes) {
//         if (annFilterByCountry.notes[ann_num].country_code == props.view.country && annFilterByCountry.notes[ann_num].risk_id == props.view.risk) {
//           annotation_dates.push(annFilterByCountry.notes[ann_num].annotation_date);
//           annotation_notes.push(annFilterByCountry.notes[ann_num].annotation_date + '\n' + annFilterByCountry.notes[ann_num].annotation);
//           annotations.push({
//             type: 'date',
//             x: annFilterByCountry.notes[ann_num].annotation_date,
//             y: 0,
//             xref: 'x',
//             yref: 'y',
//             align: 'middle',
//             valign: 'center',
//             text: '',
//             borderwidth: 0,
//             showarrow: true,
//             arrowsize: 0,
//             arrowwidth: 1,
//             arrowcolor: '#FC9F5B',
//             arrowhead: 6,
//             opacity: 0.8,
//             ax: 0,
//             ay: -200,
//           });
//         }
//         if (annFilterByCountry.notes[ann_num].country_code == 999 && annFilterByCountry.notes[ann_num].risk_id == props.view.risk) {
//           annotation_dates.push(annFilterByCountry.notes[ann_num].annotation_date);
//           annotation_notes.push(annFilterByCountry.notes[ann_num].annotation_date + '\n' + annFilterByCountry.notes[ann_num].annotation);
//           annotations.push({
//             type: 'date',
//             x: annFilterByCountry.notes[ann_num].annotation_date,
//             y: 0,
//             xref: 'x',
//             yref: 'y',
//             align: 'middle',
//             valign: 'center',
//             text: '',
//             borderwidth: 0,
//             showarrow: true,
//             arrowsize: 0,
//             arrowwidth: 1,
//             arrowcolor: '#FC9F5B',
//             arrowhead: 6,
//             opacity: 0.8,
//             ax: 0,
//             ay: -200,
//           });
//         }
//       }
//     }
//   );
//
//
//
// // Filter by ASN
// let annotation_dates = [];
// let annotation_notes = [];
// let annotations = [];
// fetch(`/static/scripts/publicAnnotation.json`)
//   .then((response) => {
//     return response.json()
//   })
//   .then((annFilterByAsn) => {
//
//     for (var ann_num in annFilterByAsn.notes) {
//       if (annFilterByCountry.notes[ann_num].asn == props.view.asn) {
//         annotation_dates.push(annFilterByAsn.notes[ann_num].annotation_date);
//         annotation_notes.push(annFilterByAsn.notes[ann_num].annotation_date + '\n' + annFilterByAsn.notes[ann_num].annotation);
//         annotations.push({
//           type: 'date',
//           x: annFilterByCountry.notes[ann_num].annotation_date,
//           y: 0,
//           xref: 'x',
//           yref: 'y',
//           align: 'middle',
//           valign: 'center',
//           text: '',
//           borderwidth: 0,
//           showarrow: true,
//           arrowsize: 0,
//           arrowwidth: 1,
//           arrowcolor: '#FC9F5B',
//           arrowhead: 6,
//           opacity: 0.8,
//           ax: 0,
//           ay: -200,
//         });
//       }
//     }
//   });
//
// // idea for a work around to turn on annotation when a marker is clicked
// //   var clickFunction = [{
// //     points: [
// //       curvernumber: 1, // index of the curve clicked on
// //       pointnumber: 1, // index of the point clicked on
// //       x: this.state.annotation_dates, // x coordinate clicked on
// //       y: this.state.annotation_dates.map(function(x) {
// //         return 0
// //       }), // y coordinate clicked on
// //       data: {}, // trace object of this curve
// //       fullData: {}, //  full trace object (that includes the defaults) of this curve
// //       xaxis: {}, // layout xaxis object corresponding to this curve
// //       yaxis: {} // '' yaxis ""
// //     ]
// //   }];
// //
// //   graphDiv.on('plotly_click', function(onClick) {
// //       /* callback goes here */
// //       plotlyData.splice(1, 0, {
// //         //x: ['2016-01-01', '2016-05-30', '2017-05-05'],
// //         x: this.state.annotation_dates,
// //         y: this.state.annotation_dates.map(function(x) {
// //           return 0
// //         }),
// //         //y: [0, 0, 0],
// //         mode: 'markers',
// //         marker: {
// //           color: 'rgba(252, 159, 91, .8)',
// //           size: 8,
// //           name: 'Annotation'
// //         },
// //         hovermode: 'closest',
// //         hoveron: 'points',
// //         hoverlabel: {
// //           bgcolor: '#FC9F5B',
// //           bordercolor: '#000000'
// //         },
// //         hoverinfo: 'text', //if `none` is set, click and hover events
// //         text: this.state.annotation_notes,
// //       });
// //
// //       state['plotlyData'] = plotlyData //.push(props.view.annotations) //push overrides the x values of the graph
// //
// //     }
// //   });
// //
// // function(onClick, getClick, getClickObj) {
// //   getClick.apply(getClickObj, [onClick])
// // });
