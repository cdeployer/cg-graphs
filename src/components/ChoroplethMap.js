import React, { Component } from 'react';
import update from 'react/lib/update'
import { connect } from 'react-redux';
import PlotlyGraph from './Plot.js';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import { fetchDataIfNeeded, riskAndDateAreSelected } from '../actions/ChoroplethMapActions';


export class ChoroplethMap extends Component {
  constructor(props) {
    super(props)
    this.state = {
      graphID: 'ChoroplethMap',
      data: [{
        type: 'choropleth',
        locationmode: 'country names',
        locations: [],
        z: [],
        autocolorscale: false,
        colorbar: {
          thickness: 5,
          len: 0.5
        },
        colorscale: [
          [0, 'rgb(231, 251, 245)'],
          [0.2, 'rgb(162, 239, 218)'],
          [0.4, 'rgb(115, 231, 199)'],
          [0.6, 'rgb(46, 219, 172)'],
          [0.8, 'rgb(0, 212, 154)'],
          [1, 'rgb(0, 174, 127)']
        ]
      }],
      layout: {
        height: 600,
        title: 'Performance of the countries around the world',
        geo: {
          showframe: false,
          showcoastlines: false,
          projection:{
            type: 'equirectangular'
          }
        }
      },
      dateToShow: this.props.view.date,
      riskToShow: this.props.view.risk
    }

    this.handleChangeRisk = this.handleChange.bind(this, 'riskToShow')
    this.handleChangeDate = this.handleChange.bind(this, 'dateToShow')
  }


  handleChange(idx, object) {
    this.setState({
      [idx]: object.value
    })
    if(idx === 'riskToShow') {
      this.props.dispatch(riskAndDateAreSelected(
        object.value,
        this.state.dateToShow
      ))
      this.props.dispatch(fetchDataIfNeeded(
        object.value,
        this.state.dateToShow
      ))
    } else if (idx === 'dateToShow') {
      this.props.dispatch(riskAndDateAreSelected(
        this.state.riskToShow,
        object.value
      ))
      this.props.dispatch(fetchDataIfNeeded(
        this.state.riskToShow,
        object.value
      ))
    }
  }


  computeState(props=this.props) {
    let locations = []
    let counts = []
    if(!props.view.isFetching && props.view.isFetched) {
      props.data[props.view.risk][props.view.date].forEach(entry => {
        if(props.countries[entry.country]) {
          locations.push(props.countries[entry.country].name)
          counts.push(entry.count)
        }
      })

      let max = Math.max(...counts)
      let z = counts.map(item => {
        return ( 100 * ( Math.log(item) / Math.log(max) ) )
      })

      return update(this.state, {
        data: [{
          locations: {$set: locations},
          z: {$set: z}
        }]
      })
    }

    return {}
  }


  componentDidMount() {
    this.props.dispatch(fetchDataIfNeeded(this.props.view.risk, this.props.view.date))
  }


  componentWillReceiveProps(nextProps) {
    if(this.props.data !== nextProps.data) {
      this.setState(this.computeState(nextProps))
    }
  }


  render() {
    const riskSelectOptions = Object.values(this.props.risks).map(risk => {
      return {
        value: risk.id,
        label: risk.title
      }
    })
    return (
      <div>
        <PlotlyGraph
          data={this.state.data}
          graphOptions={this.state.layout}
          graphID={this.state.graphID}/>
        <div className="row">
          <div className="col-sm-2 col-sm-offset-4" title="Select a date">
            <Select
              value={this.state.dateToShow}
              name='dateToShow'
              onChange={this.handleChangeDate}
              clearable={false}
              options={[{value: this.state.dateToShow, label: this.state.dateToShow}]}/>
          </div>
          <div className="col-sm-2" title="Select a risk">
            <Select
              value={this.state.riskToShow}
              name='riskToShow'
              onChange={this.handleChangeRisk}
              clearable={false}
              options={riskSelectOptions}/>
          </div>
        </div>

      </div>
    )
  }
}


const mapStateToProps = (state) => {
  return {
    view: state.ChoroplethMapViews,
    data: state.entities.cubeByRiskByDate,
    countries: state.entities.countries,
    risks: state.entities.risks
  }
}


export default connect(mapStateToProps)(ChoroplethMap);
