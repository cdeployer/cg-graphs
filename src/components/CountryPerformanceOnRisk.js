import React, { Component } from 'react';
import { connect } from 'react-redux'
import PlotlyGraph from './Plot.js';
import Select from 'react-select';
import update from 'react/lib/update'
import Highlighter from 'react-highlight-words'
import 'react-select/dist/react-select.css';
import {
  countryIsSelected, fetchDataIfNeeded,
  getCountryRanking , changeMeasure
} from '../actions/cubeActions';


export class CountryPerformanceOnRisk extends Component {
  constructor(props) {
    super(props)
    this.state = {
      cubeByRiskByCountry: {},
      graphOptions: {
        legend: {x:0, y:1},
        height: 200,
        margin: {
          l: 40,r: 30,
          b: 30,t: 0
        },
        xaxis: {
          gridcolor: 'transparent',
        },
        yaxis: {
          title: 'Count of infected devices'
        },
        font: {
          size: 9,
          color: '#7f7f7f'
        }
      },
      countries: {},
      selectorConfig: [],
      plotlyData: []
    }
  }


  computeState(props=this.props) {
    let state = {
      cubeByRiskByCountry: props.cubeByRiskByCountry,
      selectorConfig: props.view.selectorConfig
    }

    let plotlyData = []
    let lineColors = [
      'rgb(214, 39, 40)', 'rgb(31, 119, 180)', 'rgb(44, 160, 44)',
      'rgb(255, 127, 14)', 'rgb(238, 130, 238)'
    ]
    if (props.view.isFetching === 0) {
        plotlyData = props.view.selectorConfig.map(config => {
        if (config.country){
          return this.convertToPlotlySeries(
            config.country,
            props.view.risk,
            props.cubeByRiskByCountry,
            props.view.measure
          )
        }
        return {}
      }).filter(value => {return value !== undefined})
      plotlyData.forEach((trace, idx) => {
        trace['line'] = {color: lineColors[idx]}
      })
      state['plotlyData'] = plotlyData
    }

    return state
  }


  convertToPlotlySeries(countryID, riskID, cubeByRiskByCountry, measure) {
    var dataTable = cubeByRiskByCountry[riskID][countryID];
    if(dataTable) {
      return {
        x: dataTable.map(row => row.date),
        y: dataTable.map(row => row[measure] || row.count),
        name: this.props.countries[countryID].name,
        type: 'scatter',
      }
    }
  }


  componentDidMount() {
    this.props.dispatch(fetchDataIfNeeded(
      this.props.view.country,
      this.props.view.risk,
      this.props.viewId
    ))
    this.props.dispatch(getCountryRanking(
      this.props.view.country,
      this.props.view.risk,
      this.props.viewId
    ))
  };


  componentWillReceiveProps(nextProps) {
    this.setState(this.computeState(nextProps))
  }


  updateValue(idxOfSelector, selectedCountry) {
    if(selectedCountry.constructor !== Array) {
      if(!selectedCountry || selectedCountry.value === "") {
        this.props.dispatch(countryIsSelected(idxOfSelector, "", this.props.viewId))
      } else {
        this.props.dispatch(countryIsSelected(
          idxOfSelector,
          selectedCountry.value,
          this.props.viewId
        ))
        this.props.dispatch(fetchDataIfNeeded(
          selectedCountry.value,
          this.props.view.risk,
          this.props.viewId
        ))
      }
    }
  }


  buttonChange(changeEvent) {
    switch (changeEvent.target.value) {
      case 'count_normalized':
        var newState = update(this.state, {
          graphOptions: {
            yaxis: {
              title: { $set: 'Trend' }
            }
          }
        });
        this.setState(newState);
        break;
      default:
        var newState = update(this.state, {
          graphOptions: {
            yaxis: {
              title: { $set: 'Count of infected devices' }
            }
          }
        });
        this.setState(newState);

    }
    this.props.dispatch(changeMeasure(
      changeEvent.target.value,
      this.props.viewId
    ))
  }

  render() {
    return (
      <div className="graph-div">
        <form>
           <label className="radio-inline">
             <input type="radio" value="count"
               checked={this.props.view.measure === 'count'}
               onChange={this.buttonChange.bind(this)}
             />
             Simple counts
           </label>
           <label className="radio-inline">
             <input type="radio" value="count_normalized"
               checked={this.props.view.measure === 'count_normalized'}
               onChange={this.buttonChange.bind(this)}
             />
             Trend
           </label>
        </form>
        <div className="row">
          <div className="col-sm-11">
            <h3>
              {this.props.risks[this.props.view.risk].title.toUpperCase()} &nbsp; | &nbsp;
              {this.props.countries[this.props.view.country].name.toUpperCase()}
            </h3>
          </div>
          <div className="col-sm-1">
            { this.props.view.rank ? <a href="/country" title="Rank of country on this risk in last month with #1 = worst"><h3 className="pull-right">#{this.props.view.rank}</h3></a> : '' }
          </div>
        </div>
        <PlotlyGraph
          data={this.state.plotlyData}
          graphOptions={this.state.graphOptions}
          graphID={this.props.viewId} />
        {this.state.selectorConfig.map((selectInfo, idx) => {
          return <CountrySelect
                    countries={Object.values(this.props.countries)}
                    disabled={selectInfo.disabled}
                    onChange={this.updateValue.bind(this, idx)}
                    selectedCountry={selectInfo.country}
                    key={idx}
                    />
        })}
      </div>
    );
  }
}


export class CountrySelect extends Component {
  constructor(props) {
    super(props)
    this.state= {
      inputValue: ''
    }
  }

  optionRenderer(option) {
    if(!option.label){ return }
    return (
      <Highlighter
        searchWords={[this.state.inputValue]}
        textToHighlight={option.label}
      />
    );
  }


  setInputValue(value) {
    this.setState({
      inputValue: value
    })
  }

  render() {
    const selectOptions = this.props.countries.map(country => {
      return {
        value: country.id,
        label: country.name
      }
    })
    selectOptions.unshift({value: '', label: 'Select a country'})
    return (
      <div className="Select-div">
        <Select
          name="countries"
          value={this.props.selectedCountry || selectOptions[0]}
          options={selectOptions}
          onChange={this.props.onChange}
          onInputChange={this.setInputValue.bind(this)}
          optionRenderer={this.optionRenderer.bind(this)}
          disabled={this.props.disabled}
          clearable={false}
        />
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    cubeByRiskByCountry: state.entities.cubeByRiskByCountry,
    countries: state.entities.countries,
    risks: state.entities.risks
  }
}


export default connect(mapStateToProps)(CountryPerformanceOnRisk)
