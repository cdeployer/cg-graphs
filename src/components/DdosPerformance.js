import React, { Component } from 'react';
import { connect } from 'react-redux';
import PlotlyGraph from './Plot.js';
import { fetchDataIfNeeded } from '../actions/cubeActions';
//front page graph
export class DdosPerformance extends Component {
  constructor(props) {
    super(props)
    this.state = {
      graphOptions: {
        height: 600,
        xaxis: {
         gridcolor: 'transparent',
        },
        yaxis: {
         title: this.props.view.yLabel
        },
        barmode: 'stack',
        annotations: [{
          xref: 'paper',
          yref: 'paper',
          x: 1.056,
          xanchor: 'left',
          y: 0.99,
          yanchor: 'bottom',
          text: 'Risks',
          legendtitle: true,
          showarrow: false
        }]
      },
      plotlyData: []
    }
  }


  computeState(props=this.props) {
    let state = {}
    let plotlyData = []
    let barColors = [
      'rgb(200, 2, 16)', 'rgb(0, 212, 154)',
      'rgb(84, 114, 222)', 'rgb(96, 3, 212)'
    ]

    if (props.view.isFetched) {
      plotlyData = props.view.risksToView.map((risk, idx) => {
        return this.convertToPlotlySeries(
          props.view.country,
          risk,
          props.data,
          props.view.measure,
          barColors[idx]
        )
      }).filter(value => {return value !== undefined})

      state['plotlyData'] = plotlyData
    }

    return state
  }
  // this code should be implimented somewhere above.
  // plotlyData.splice(1, 0, {
  //   //x: ['2016-01-01', '2016-05-30', '2017-05-05'],
  //   x: this.state.annotation_dates,
  //   y: this.state.annotation_dates.map(function (x){ return 0}),
  //   //y: [0, 0, 0],
  //   mode: 'markers',
  //   marker: {
  //     color: 'rgba(252, 159, 91, .8)', size: 8,
  //   name: 'Annotation'},
  //   hovermode:'y',
  //   hoverlabel:{
  //     bgcolor: '#FC9F5B',
  //     bordercolor: '#000000'},
  //   hoverinfo:'text',
  //   text: this.state.annotation_notes,
  // });
  //
  // /* Write up more console logs to pinpoint where data is created */
  // console.log('plotlyData view', props.view.annotations)
  // console.log('plotlyData', plotlyData)
  // state['plotlyData'] = plotlyData //.push(props.view.annotations) //push overrides the x values of the graph
  // console.log('state plotlyData', state['plotlyData'])
  // }

  convertToPlotlySeries(countryID, risk, cubeByRiskByCountry, measure, color) {
    var dataTable = cubeByRiskByCountry[risk.id][countryID];
    if(dataTable) {
      return {
        x: dataTable.map(row => row.date),
        y: dataTable.map(row => (row[measure] || row.count )/ 1000000),
        name: risk.title,
        type: 'bar',
        marker: {color: color}
      }
    }
  }


  componentDidMount() {
    this.props.view.risksToView.forEach(risk =>{
      this.props.dispatch(fetchDataIfNeeded(
        this.props.view.country,
        risk.id,
        this.props.view.id,
        'DdosPerformanceViews'
      ))
    })
  };


  componentWillReceiveProps(nextProps) {
    this.setState(this.computeState(nextProps))
  }


  render() {
    return (
      <div>
        <PlotlyGraph
          data={this.state.plotlyData}
          graphOptions={this.state.graphOptions}
          graphID={this.props.view.id}/>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    view: state.DdosPerformanceViews['global/ddos'],
    data: state.entities.cubeByRiskByCountry
  }
}


export default connect(mapStateToProps)(DdosPerformance);
