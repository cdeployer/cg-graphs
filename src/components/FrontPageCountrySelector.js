import React, { Component } from 'react';
import Highlighter from 'react-highlight-words'
import Select from 'react-select';
import Loader from 'halogen/BounceLoader'
import 'react-select/dist/react-select.css';

export default class FrontPageCountrySelector extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selectedCountry: undefined,
      inputValue: '',
      spinner: '0px'
    }
  }


  onChange(country) {
    // purely for testing atm - we don't use state internally
    this.setState({
      selectedCountry: country,
      spinner: '40px'
    });
    if (country.slug) {
      window.location = `/country/${country.slug}`
    }
  }


  optionRenderer(option) {
    if(!option.label){ return }
    return (
      <Highlighter
        searchWords={[this.state.inputValue]}
        textToHighlight={option.label}
      />
    );
  }


  setInputValue(value) {
    this.setState({
      inputValue: value
    })
  }

  render() {
    const selectOptions = Object.values(this.props.countries).map(country => {
      return {
        value: country.id,
        label: country.name,
        slug: country.slug
      }
    })

    selectOptions.unshift({value: '', label: 'Select a country'})
    return (
      <div>
        <Select
          name="country-selector"
          value={this.state.selectedCountry || selectOptions[0]}
          options={selectOptions}
          onChange={this.onChange.bind(this)}
          onInputChange={this.setInputValue.bind(this)}
          optionRenderer={this.optionRenderer.bind(this)}
        />
        <Loader size={this.state.spinner} />
      </div>
    );
  }
}
