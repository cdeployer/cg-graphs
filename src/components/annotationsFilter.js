import React, {
    Component
} from 'react';
import {
    connect
} from 'react-redux'
import PlotlyGraph from './Plot.js';
import Select from 'react-select';
import update from 'react/lib/update'
import Highlighter from 'react-highlight-words'
import 'react-select/dist/react-select.css';
//need to move file to this location
import '/api/publicAnnotation.json';
import {
    countryIsSelected,
    fetchDataIfNeeded,
    getCountryRanking,
    changeMeasure
} from '../actions/cubeActions';

let annotation_dates = [];
let annotation_notes = [];
let annotations = [];
let annotationFormat = [{
    type: 'date',
    x: annData.notes[ann_num].annotation_date,
    y: 0,
    xref: 'x',
    yref: 'y',
    align: 'middle',
    valign: 'center',
    text: '',
    borderwidth: 0,
    showarrow: true,
    arrowsize: 0,
    arrowwidth: 1,
    arrowcolor: '#FC9F5B',
    arrowhead: 6,
    opacity: 0.8,
    ax: 0,
    ay: -200,
}];

//filter annotation Country Comparison
function annotationCountryComparisonFetchAndFilter(risk_id, country_code, asn) {
    return (
        fetch(`/static/scripts/publicAnnotation.json`)
        //change to /api/publicAnnotation.json for staging
        .then((response) => {
            return response.json()
        })

        .then((annData) => {
            console.log(annData);
            for (var ann_num in annData.notes) {
                console.log(props.view);
                if (annData.notes[ann_num].risk_id == risk_id) {
                    console.log(ann_num);
                    console.log(annData.notes[ann_num].annotation_date);
                    annotation_dates.push(annData.notes[ann_num].annotation_date);
                    annotation_notes.push(annData.notes[ann_num].annotation_date + '\n' + annData.notes[ann_num].annotation);
                    annotations.push({
                        return(annotationFormat)
                    })
                }
            }
        }));
}

//filter annotation AS Source
function annotationAsSourceFilter(props) {
    return (
        fetch(`/static/scripts/publicAnnotation.json`)
        //change to /api/publicAnnotation.json for staging
        .then((response) => {
            return response.json()
        })

        .then((annData) => {
            console.log(annData);
            for (var ann_num in annData.notes) {
                console.log(props.view);
                if (annData.notes[ann_num].risk_id == props.view.risk) {
                    console.log(ann_num);
                    console.log(annData.notes[ann_num].annotation_date);
                    annotation_dates.push(annData.notes[ann_num].annotation_date);
                    annotation_notes.push(annData.notes[ann_num].annotation_date + '\n' + annData.notes[ann_num].annotation);
                    annotations.push({
                        return(annotationFormat)
                    })
                }
            }
        }));
}
