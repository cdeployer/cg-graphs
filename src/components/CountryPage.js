import React, { Component } from 'react';
import { connect } from 'react-redux';
import CountryPerformanceOnRisk from './CountryPerformanceOnRisk';


export class CountryPage extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        {Object.keys(this.props.views).map((key) => {
          return <CountryPerformanceOnRisk view={this.props.views[key]} key={key} id={key}/>
        })}
      </div>
    );
  }
}


const mapStateToProps = (state, ownProps) => {
  return {
    views: state.countryPerformanceOnRiskViews
  }
}


export default connect(mapStateToProps)(CountryPage)
