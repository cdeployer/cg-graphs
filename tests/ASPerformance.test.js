import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {ASPerformance} from '../src/components/ASPerformance';

describe('ASPerformance component is working as expected', () => {
  function setup() {
    let props = {
      asn: {
        174: {
          id: 174,
          name: 'ASN 174'
        }
      },
      view: {
        id: 'GB/1/174',
        as: 174,
        risk: 1,
        country: 'GB',
        type: 'as/performance',
        isFetched: false,
        isFetching: false,
        didFailed: false,
        measure: "count_normalized",
        selectorConfig: [
          {disabled: true, as: 174},
          {disabled: true, as: 0},
          {disabled: false, as: undefined},
          {disabled: false, as: undefined},
          {disabled: false, as: undefined}
        ]
      }
    }
    const enzymeWrapper = shallow(<ASPerformance {...props} />)

    return {
      props,
      enzymeWrapper
    }
  }

  it('renders expected DOM', () => {
    const { enzymeWrapper } = setup()
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  })
})
