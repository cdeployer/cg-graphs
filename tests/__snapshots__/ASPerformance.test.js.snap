exports[`ASPerformance component is working as expected renders expected DOM 1`] = `
<div
  style={
    Object {
      "marginBottom": "80px",
    }
  }>
  <PlotlyGraph
    data={Array []}
    graphOptions={Object {}} />
  <ASSelect
    asn={
      Array [
        Object {
          "id": 174,
          "name": "ASN 174",
        },
      ]
    }
    disabled={true}
    onChange={[Function]}
    selectedAS={174} />
  <ASSelect
    asn={
      Array [
        Object {
          "id": 174,
          "name": "ASN 174",
        },
      ]
    }
    disabled={true}
    onChange={[Function]}
    selectedAS={0} />
  <ASSelect
    asn={
      Array [
        Object {
          "id": 174,
          "name": "ASN 174",
        },
      ]
    }
    disabled={false}
    onChange={[Function]} />
  <ASSelect
    asn={
      Array [
        Object {
          "id": 174,
          "name": "ASN 174",
        },
      ]
    }
    disabled={false}
    onChange={[Function]} />
  <ASSelect
    asn={
      Array [
        Object {
          "id": 174,
          "name": "ASN 174",
        },
      ]
    }
    disabled={false}
    onChange={[Function]} />
</div>
`;
